### DISCUSSION TITLE
Attributes on primary constructors (records)

### DISCUSSION THREAD

**Discussion by raffaeler** [dotnet/csharplang#5128] [Q&A] [answered]
I had a very annoying issue using `System.Text.Json` and records: https://github.com/dotnet/runtime/issues/45373

_The main cause_ is that primary constructors cannot be annotated with an attribute because they are intended to be used on constructors and not type declarations.

BTW This issue is hitting a project so hard that we are thinking to write a custom serializer to support the multitude of types impacted. The proposed solution is not acceptable.

_The alternative solution_ is changing the compiler to allow `AttributeTargets.Constructor` annotating the type but targeting the primary constructors in order to preserve the semantics.

Thoughts?

**Comment by svick** [comment, marked as answer]
This is a duplicate of https://github.com/dotnet/csharplang/discussions/3650.

**Reply by raffaeler** [reply]
Sorry, missed that
